### PR TITLE
memory pair game

### PR DESCRIPTION
**Pull request by DaniaB24**
### memory pair game

[Demo](https://daniab24.github.io/memory-pair-game/) |
[Code base](https://github.com/DaniaB24/memory-pair-game)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
Please keep only files required for the code review (i.e. codebase part that was authored by you).
See section B123 of [submission guidelines](https://github.com/kottans/frontend-2022-homeworks/blob/main/README.md). This helps reviewers jump straight to their job and makes their life easier.

**Comment by DaniaB24** [comment]
> Please keep only files required for the code review (i.e. codebase part that was authored by you). See section B123 of [submission guidelines](https://github.com/kottans/frontend-2022-homeworks/blob/main/README.md). This helps reviewers jump straight to their job and makes their life easier.

Hi^ sorry, okey I will remove)

**Comment by OleksiyRudenko** [comment]
Quite unexpected UX, esp. closer to the end of game.
Also doesn't reshuffle cards for the next game.
Record_2022_08_17_14_37_34_731-xs

**Comment by DaniaB24** [comment]
> Quite unexpected UX, esp. closer to the end of game.
> Also doesn't reshuffle cards for the next game.
> Record_2022_08_17_14_37_34_731-xs

okey, I will do it^ thanks for the review <3

**Comment by OleksiyRudenko** [comment]
Hwo about posting a link in the students' chat and collecting user feedback?

**Comment by DaniaB24** [comment]
> Looks visually nice.
> How about posting a link in the students' chat and collecting user feedback?

okey, thanks a lot

**Comment by OleksiyRudenko** [comment]
This PR contains code from different tasks. Need to fix this

**Comment by DaniaB24** [comment]
> This PR contains code from different tasks. Need to fix this

yeah... I did it

**Comment by OleksiyRudenko** [comment]
Which files do we review?

image

**Comment by DaniaB24** [comment]
> Which [files](https://github.com/kottans/frontend-2019-p2p/pull/198/files) do we review?
> 
> image

@OleksiyRudenko  re-review pls:)  i think now everything is clear^

**Comment by DaniaB24** [comment]
@OleksiyRudenko

**Comment by DaniaB24** [comment]
@OleksiyRudenko  re-review please) thank you a lot

**Comment by DaniaB24** [comment]
@OleksiyRudenko

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    firstCard = targetCard;
+    firstCard.classList.add("flip");
+    return;
+  } else {
+    secondCard = targetCard;
+    secondCard.classList.add("flip");
+    checkMatch();
+  }
+}
+const memoryCards = [...allPlayerCard, ...allPlayerCard];
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:52]: Move this var next to other vars.

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      <img src=${item.imgBack} class="back_card">
+      <img src=${item.imgFront} class="front_card">
+    </div>
+`;
+  });
+  clonedCard.forEach((item) => {
+    allCard.insertAdjacentHTML("beforeend", item);
+  });
+}
+createCard();
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:73]: Move init function to the bottom of file.

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    checkMatch();
+  }
+}
+const memoryCards = [...allPlayerCard, ...allPlayerCard];
+function sortCard() {
+  memoryCards.sort(function () {
+    return 0.5 - Math.random();
+  });
+}
+function createCard() {
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:58]: Looks like it is not about one card, it is about one card deck.

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  sortCard();
+
+  let clonedCard = memoryCards.map((item) => {
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:61]: Should be plural. And why do you use let here?

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  });
+  clonedCard.forEach((item) => {
+    allCard.insertAdjacentHTML("beforeend", item);
+  });
+}
+createCard();
+
+function checkMatch() {
+  if (firstCard.id === secondCard.id) {
+      disableCard();
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:77]: Check your editor config. You should see redundant tab/spaces/empty lines.

File: submissions/DaniaB24/memory-pair-game/js/app.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    secondCard.classList.add("hide");
+  }, 800);
+}
+function freezeCard() {
+  lockGame = true;
+  setTimeout(() => {
+    lockGame = false;
+    firstCard.classList.remove("flip");
+    secondCard.classList.remove("flip");
+    firstCard = "";
```
Comment (lysenko-sergey-developer) [inline submissions/DaniaB24/memory-pair-game/js/app.js:100]: Resetting to null will be better.

**Review by lysenko-sergey-developer — CHANGES_REQUESTED** [review]

**Comment by lysenko-sergey-developer** [comment]
@DaniaB24 Good work!👍

**Comment by DaniaB24** [comment]
> @DaniaB24 Good work!👍

thank you a lot for review 🤝🤗 I will do it🫡

**Comment by DaniaB24** [comment]
@lysenko-sergey-developer @OleksiyRudenko friendly reminder

**Comment by DaniaB24** [comment]
@OleksiyRudenko  @lysenko-sergey-developer  thanks a lot<333

**Comment by OleksiyRudenko** [comment]
@DaniaB24 looks like you are missing only submission for [OOP Exercise](https://github.com/kottans/frontend/blob/master/tasks/js-post-oop.md) to have the full house of tasks.

**Comment by DaniaB24** [comment]
> @DaniaB24 looks like you are missing only submission for [OOP Exercise](https://github.com/kottans/frontend/blob/master/tasks/js-post-oop.md) to have the full house of tasks.

https://github.com/kottans/frontend-2022-homeworks/pull/172
it seems to be here

**Comment by OleksiyRudenko** [comment]
> [kottans/frontend-2022-homeworks#172](https://github.com/kottans/frontend-2022-homeworks/pull/172) it seems to be here

That PR covers [Building a Tiny JS World](https://github.com/kottans/frontend/blob/master/tasks/js-pre-oop.md)

**Comment by DaniaB24** [comment]
> > [kottans/frontend-2022-homeworks#172](https://github.com/kottans/frontend-2022-homeworks/pull/172) it seems to be here
> 
> That PR covers [Building a Tiny JS World](https://github.com/kottans/frontend/blob/master/tasks/js-pre-oop.md)

image
but there is a code for oop 🤦‍♂️

**Comment by DaniaB24** [comment]
> > [kottans/frontend-2022-homeworks#172](https://github.com/kottans/frontend-2022-homeworks/pull/172) it seems to be here
> 
> That PR covers [Building a Tiny JS World](https://github.com/kottans/frontend/blob/master/tasks/js-pre-oop.md)



> > > [kottans/frontend-2022-homeworks#172](https://github.com/kottans/frontend-2022-homeworks/pull/172) it seems to be here
> > 
> > 
> > That PR covers [Building a Tiny JS World](https://github.com/kottans/frontend/blob/master/tasks/js-pre-oop.md)
> 
> image but there is a code for oop 🤦‍♂️

sorry, my bad

**Pull request merged**